packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -706,51 +704,45 @@ export class EditManager<
-		for (const newCommit of newCommits) {
-			// Step 1 - Recreate the peer remote client's local environment.
-			// Get the revision that the remote change is based on
-			const [, baseRevisionInTrunk] = this.getClosestTrunkCommit(referenceSequenceNumber);
-			// Rebase that peer local branch over the part of the trunk up to the base revision
-			// This will be a no-op if the sending client has not advanced since the last time we received an edit from it
-			const peerLocalBranch = getOrCreate(
-				this.peerLocalBranches,
-				sessionId,
-				() =>
-					new SharedTreeBranch(baseRevisionInTrunk, this.changeFamily, this.mintRevisionTag),
-			);
-			peerLocalBranch.rebaseOnto(this.trunk, baseRevisionInTrunk);
+		// Step 1 - Recreate the peer remote client's local environment.
+		// Get the revision that the remote change is based on and rebase that peer local branch over the part of the
+		// trunk up to the base revision. This will be a no-op if the sending client has not advanced since the last
+		// time we received an edit from it
+		const [, baseRevisionInTrunk] = this.getClosestTrunkCommit(referenceSequenceNumber);
+		const peerLocalBranch = getOrCreate(
+			this.peerLocalBranches,
+			sessionId,
+			() => new SharedTreeBranch(baseRevisionInTrunk, this.changeFamily, this.mintRevisionTag),
+		);
+		peerLocalBranch.rebaseOnto(this.trunk, baseRevisionInTrunk);
 
-			// Step 2 - Append the change to the peer branch. Rebase the change to the tip of the trunk.
-			if (peerLocalBranch.getHead() === this.trunk.getHead()) {
-				// If the branch is fully caught up and empty after being rebased, then push to the trunk directly
+		// Step 2 - Append the changes to the peer branch and rebase the changes to the tip of the trunk.
+		if (peerLocalBranch.getHead() === this.trunk.getHead()) {
+			// If the peer local branch is fully caught up and empty (no changes relative to the trunk) after being
+			// rebased, then push changes to the trunk directly and update the peer branch to the trunk's head.
+			for (const newCommit of newCommits) {
 				this.pushCommitToTrunk(nextSequenceId, { ...newCommit, sessionId });
-				peerLocalBranch.setHead(this.trunk.getHead());
-			} else {
-				// Otherwise, rebase the change over the trunk and append it, and append the original change to the peer branch.
-				const { duration, output: newChangeFullyRebased } = measure(() =>
-					rebaseChange(
-						this.changeFamily.rebaser,
-						newCommit,
-						peerLocalBranch.getHead(),
-						this.trunk.getHead(),
-						this.mintRevisionTag,
-					),
-				);
-
-				this.telemetryEventBatcher?.accumulateAndLog({
-					duration,
-					...newChangeFullyRebased.telemetryProperties,
-				});
-
+				nextSequenceId = getNextSequenceId(nextSequenceId);
+			}
+			peerLocalBranch.setHead(this.trunk.getHead());
+		} else {
+			// Otherwise, push the changes to the peer local branch and merge the branch over the trunk.
```

Comment:
We typically use "rebase over" but "merge into". Not sure how those conventions came to be. 🤷‍♂️
```suggestion
			// Otherwise, push the changes to the peer local branch and merge the branch into the trunk.
```